### PR TITLE
[ZEPPELIN-4226] Fix "View in Spark web UI" in kubernetes mode

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
When Zeppelin is running in Kubernetes, "View in Spark web UI" gives internal address, instead of address defined in SERVICE_DOMAIN.

I think this problem is side effect of https://github.com/apache/zeppelin/pull/3375 and this PR includes fix and updated unittest.

### What type of PR is it?
Bug Fix

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-4226

### How should this be tested?
Run Zeppelin on kubernetes, and run spark job, click "View in Spark web UI" button.

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala [hunk 1]
```diff
@@ -303,7 +303,11 @@ abstract class BaseSparkScalaInterpreter(val conf: SparkConf,
 
   protected def createZeppelinContext(): Unit = {
     val sparkShims = SparkShims.getInstance(sc.version, properties)
-    sparkShims.setupSparkListener(sc.master, sparkUrl, InterpreterContext.get)
+    var webUiUrl = properties.getProperty("zeppelin.spark.uiWebUrl");
+    if (webUiUrl == null || webUiUrl.trim().length() == 0) {
```
Comment (zjffdu) [inline spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala]: Maybe use `StringUtils.isBlank()` which is more clean

Comment (Leemoonsoo) [reply spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala]: Updated!

**Comment by zjffdu** [comment]
LGTM

**Comment by Leemoonsoo** [comment]
Merge to master

**Pull request closed without merging**